### PR TITLE
test: cover the remaining policy branches and document their edge cases

### PR DESCRIPTION
**Pull request by remedcu**
Close the coverage gaps left in the policies, and state in NatSpec the four behaviours a reader would otherwise have to infer.

## Context and Motivation

With test-only contracts excluded, production coverage read 99.31% of statements and 94% of branches. The shortfall was a handful of decode guards and fail-closed paths, each of which is exactly the sort of thing an audit asks about. This brings statements, branches and lines to 100%.

The pre-audit review also turned up three documented-nowhere behaviours, all fail-closed or deliberate rather than bugs, and a fourth worth stating beside them. They are recorded here alongside the tests, since both answer the same question about the same contracts.

## Changes

Tests:
- `InvalidMultiSend`, `InvalidApproval` and `InvalidTransfer`: the policies defending their own decode.
- `NoCosignerConfigured`: configuring a zero-address co-signer must fail closed at check time rather than accept any signature.
- `getCoSigner` and `isSpenderAllowed`.
- Two `_readGuardSlot` cases, with `TestNonSafeConfigurer` and `TestRevertingStorageConfigurer`.

NatSpec:
- `CoSignerPolicy`: transaction path only; the `nonce() - 1` derivation assumes an increment `execTransactionFromModule` never performs.
- `NativeTransferPolicy`: the recipient comes from the access selector, and the CALL fallback key permits any amount to any address.
- `ERC20ApprovePolicy`: `approve(spender, 0)` bypasses the allowlist by design.
- `ERC20TransferPolicy`: `transferFrom`'s `from` is deliberately unconstrained.

`.solcover.js` additionally skips `libraries/SignatureExtension.sol`.

## Decisions and Tradeoffs

- The three decode-guard tests call `checkTransaction` directly. The engine keys on the selector, and each policy's `configure` pins which selectors it accepts, so mismatched calldata can never be routed to these policies through a Safe. A direct call is the only thing that reaches the guard, and the policies are permissionless by design.
- `SignatureExtension` is vendored from `safe-research/safenet` -- differing only in the pragma and the SPDX identifier -- which owns its unit tests in `contracts/test/libraries/SignatureExtension.t.sol` -- including both standalone rejection paths. Rather than duplicate them, the file is skipped for coverage with a comment pointing at the upstream tests. It is still exercised end-to-end here through the guard's context decoding.
- The first `_readGuardSlot` test is not sufficient on its own. A plain non-Safe caller reverts with empty return data, which the `returnData.length < 96` check already rejects, so that test still passes with the `!success` condition removed. `TestRevertingStorageConfigurer` reverts with a 96-byte payload shaped like a real answer, planting the guard's own address in the word the guard reads; only the call status separates the two. That test does fail without `!success`.
- The `returnData.length < 96` half of the same condition is deliberately left unpinned. Dropping it is an equivalent mutant: a short answer makes the assembly read past the allocation, and unallocated memory is zero-initialised, so the read yields the same `address(0)` the length check would have returned. Making it observable would need a 65-95 byte answer *and* a guard address ending in `0x00`, and even then the only effect is that the caller blocks its own configuration. The check is defence-in-depth against the over-read; a test asserting it would pass with or without it.
- Coverage now reports one uncovered function, `PolicyEngine._allowedCalls`. Its default body is empty and every engine here overrides it without calling `super`, so the base implementation is unreachable. Left untested and undocumented in the contract, since a note about a coverage artifact does not belong in NatSpec.
- The NatSpec edits leave executable bytecode identical but change the appended solc metadata, so `keccak256(initcode)` -- and therefore the CREATE2 address a deterministic deploy produces -- moves for exactly these four policies. Anyone holding a bytecode or metadata hash for them will see it change from a diff that reads as comment-only.
- Solhint enforces one contract per file, hence two test-contract files.

## Testing

Suite 128 passing, up from 120; build, lint and typecheck green. Coverage is 100% of statements, branches and lines on production contracts. Two of the new tests were checked against a temporarily reverted contract to confirm they fail without the behaviour they cover -- `InvalidTransfer` and the reverting-probe case -- after the second was found to pass regardless on its first attempt.

### CODE REVIEW

File: test/erc20TransferPolicy.spec.ts [hunk 1]
```diff
@@ -331,5 +331,24 @@ describe('ERC20TransferPolicy', function () {
...
+      // `configure` pins the selector to one of the two, so the engine can only route those here.
+      // This is the policy defending its own decode when called directly, which anyone may do.
+      const { safe, erc20TransferPolicy, token } = await loadFixture(fixture)
+
+      await expect(
+        erc20TransferPolicy.checkTransaction(
+          safe,
+          token,
+          0n,
+          token.interface.encodeFunctionData('approve', [randomAddress(), 1n]),
```
Comment (akshay-ap) [inline test/erc20TransferPolicy.spec.ts]: nit: can be a random selector other than transfer and transferFrom

File: test/erc20ApprovePolicy.spec.ts [hunk 2]
```diff
@@ -290,5 +290,48 @@ describe('ERC20ApprovePolicy', function () {
...
+      // `configure` pins the selector, so the engine can only route `approve` calldata here. This is
+      // the policy defending its own decode when called directly, which anyone may do.
+      const { safe, erc20ApprovePolicy, token } = await loadFixture(fixture)
+
+      await expect(
+        erc20ApprovePolicy.checkTransaction(
+          safe,
+          token,
+          0n,
+          token.interface.encodeFunctionData('transfer', [randomAddress(), 1n]),
```
Comment (akshay-ap) [inline test/erc20ApprovePolicy.spec.ts]: nit: can be a random selector other than approve

**Review by akshay-ap — APPROVED** [review]
